internal/repository/hook.go

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package repository
+
+import (
+	"fmt"
+	"os"
+	"path"
+)
+
+type HookType string
+
+var HookPrePush = HookType("pre-push")
+
+func (r *Repository) UpdateHook(hookType HookType, content []byte) error {
+	conf, err := r.r.Config()
+	if err != nil {
+		return fmt.Errorf("reading repository config: %w", err)
+	}
+
+	repoRoot := conf.Core.Worktree
+	hookFile := path.Join(repoRoot, ".git", "hooks", string(hookType))
```

Comment:
It'd be nice to identify `.git` rather than hardcode it, for cases with detached git-dirs.

Reply:
I am not sure what a "detached git-dir" is. Could you provide me an example so I can test against the implementation?

Reply:
oh wow, TIL! 

Do you know if there is a better approach than using `r.r.Worktree()` to eventually get to the `hooks` folder? If not then I will add more logic to test if `.git` is dir or file etc...

Reply:
Created issue (https://github.com/go-git/go-git/issues/977) and added comment.